packages/varlock-website/src/content/docs/integrations/cloudflare.mdx

```diff
@@ -0,0 +1,219 @@
+---
+title: Cloudflare Workers
+description: How to integrate varlock with Cloudflare Workers and Wrangler for secure, type-safe environment management
+---
+import SyncedCodeTabs from '@/components/SyncedCodeTabs.astro'
+import { TabItem, Tabs, Steps } from "@astrojs/starlight/components";
+import Badge from '@/components/Badge.astro';
+import ExecCommandWidget from '@/components/ExecCommandWidget.astro';
+import InstallJsDepsWidget from '@/components/InstallJsDepsWidget.astro';
+
+<div class="page-badges">
+  <Badge npmPackage="@varlock/vite-integration" />
+</div>
+
+Varlock provides a robust solution for managing environment variables in Cloudflare Workers, offering validation, type safety, and security features that go beyond Cloudflare's built-in environment variable handling.
+
+
+## Two Approaches
+
+There are two main ways to use varlock with Cloudflare Workers:
+
+1. **With Vite plugin** (recommended) - Use the Vite integration alongside the Cloudflare Workers Vite plugin
+2. **Without Vite** - Use Wrangler's `vars` and `secrets` directly
+
+## Approach 1: Using the Vite Plugin (Recommended)
+
+The Vite plugin bundles resolved environment variables into your built code, making it safe and straightforward to use.
+
+:::note[Backend-only Workers]
+Even for backend-only Workers, using the Vite plugin is recommended. Because it bundles the resolved env into your built code, within the Cloudflare dashboard, your team members can view the source code. Be mindful of this when including sensitive values or sending source maps to external services.
+:::
+
+### Setup
+
+<Steps>
+
+1. **Install varlock and the Vite integration package**
+    <InstallJsDepsWidget packages="@varlock/vite-integration varlock" />
+
+1. **Run `varlock init` to set up your `.env.schema` file**
+
+    This will guide you through setting up your `.env.schema` file, based on your existing `.env` file(s). Make sure to review it carefully.
+
+    <ExecCommandWidget command="varlock init" showBinary={false} />
+
+1. **Enable the Vite config plugin**
+
+    Add both the varlock Vite plugin and the Cloudflare Workers Vite plugin to your `vite.config.*` file:
+
+    ```diff lang="ts" title="vite.config.ts"
+    import { defineConfig } from 'vite';
+    +import { varlockVitePlugin } from '@varlock/vite-integration';
+
+    export default defineConfig({
+    -  plugins: [otherPlugin()],
+    +  plugins: [
+    +    varlockVitePlugin(),
+    +    cloudflare(),
+    +    otherPlugin()
+    +  ],
+    });
+    ```
+
+    The varlock plugin will automatically detect Cloudflare Workers and use the `resolved-env` mode, which injects the fully resolved environment data into your built code.
+
+</Steps>
+
+### Local Development
```

Comment:
this stuff is only needed for the non vite way!

From further reading, I guess there are actually 3 ways.

- use vite
- inject the whole env and use our `ENV` object
- use the `"env -i` trick and dont use our `ENV` object